client/extra.go

```diff
@@ -6,14 +6,19 @@ import (
 	"github.com/ovrclk/akash/x/market/types"
 )
 
-// TODO: implement with search parameters
-
+// ActiveLeasesForProvider filter the Lease search to active state and then
+// provider's ID AccAddress.
 func (c *qclient) ActiveLeasesForProvider(id sdk.AccAddress) (mquery.Leases, error) {
-	leases, err := c.Leases(mquery.LeaseFilters{})
+	// NOTE: Currently unable to filter by Provider in tendermint scan
+	lf := mquery.LeaseFilters{
+		// TODO: Add filters when tests prove functionality
+	}
+	leases, err := c.Leases(lf)
 	if err != nil {
 		return nil, err
 	}
 
+	// TODO: Secondary filtering *should* not be necessary
```

Comment:
Just leave this stuff as it was before - these changes don't really add anything.

refs #604 #725

Reply:
:heavy_check_mark: